utils/src/math.rs

```diff
@@ -17,9 +19,45 @@ pub fn ceil_log2(d: usize) -> usize {
     ceil_log2
 }
 
+/// Evaluates a polynomial's `coefficients` at a given point `x`, using [Horner's scheme](https://en.wikipedia.org/wiki/Horner%27s_method)
+pub fn evaluate_polynomial<F, I>(coefficients: I, zero: F, x: F) -> F
+where
+    I: Iterator<Item = F>,
+    F: Mul<F, Output = F> + Add<F, Output = F> + Copy,
+{
+    coefficients.fold(zero, |acc, coeff| x * acc + coeff)
+}
```

Comment:
I think this is not going to work as you're going through the coeffs in the wrong order no?
Also looking at this, I realize that we have a very similar function in `ExtendedDensePolynomial::eval_polynomial`, perhaps we can just use that?

Reply:
> coeffs in the wrong order 

interesting, because I am not sure in what order the coeffs usually would be (especially in the `DensePolynomial` implementation) - I assumed they would be listed "from left to right", so `[2, 6, 2, 1]` would correspond to `2x³ + 6x² + 2x + 1`, I guess in this case they are handled the other way around natively? I was confused at first when I saw that the coeffs were reversed in the already existing implementations

> I realize that we have a very similar function in ExtendedDensePolynomial::eval_polynomial, perhaps we can just use that?

I did notice the implementation of `eval_polynomial` as well, assuming most calculations are done using the `ExtendedDensePolynomial` trait, the additional implementation could just be ignored

Reply:
yeah they are handled from left to right as far as I remember

Reply:
Do you want to try replacing occurrences of this pattern with the `eval_polynomial` function? Sorry, didn't realize it was already there ^^

Reply:
also I realize that we might just be able to do this:

```rust
DensePolynomial::<F>::from_coefficients_slice(&coeffs).evaluate(x);
```

I bet arkworks `evaluate` function is implemented like this (we should verify though)

Reply:
will look into it 👍

Reply:
@mimoo ye, arkwork's implementation of `evaluate` also uses horner's scheme, but more efficiently in parallel - so probably better to use that. I will go through the code and see if it can be replaced somewhere